src/test/steps/signUpSteps.ts

```diff
@@ -0,0 +1,38 @@
+import { When, Then } from '@cucumber/cucumber';
+import basePage from '../../pages/basePage';
+
+let randomNum;
+Then('User clicks {string} on {string}', async function (button, page) {
+  const [buttonName, pageName] = basePage.wrapper.toCamelCase(button, page);
+  await basePage.wrapper.waitAndClick(basePage[pageName][buttonName]);
+});
+Then('User fills {string} as {string} on {string}', async function (placeholder, input, page) {
+  const [placeholderName, pageName] = basePage.wrapper.toCamelCase(placeholder, page);
+  await basePage.wrapper.type(basePage[pageName][placeholderName], input);
+});
+
+Then('User enters {string} as {string} on {string}', async function (email, input, page) {
+  if (!randomNum) {
+    randomNum = Math.floor(Math.random() * 10000);
+  }
+  const [pageName, placeholderName] = basePage.wrapper.toCamelCase(page, email);
+  const updatedInput = `${input}${randomNum}@gmail.com`;
```

Comment:
Do you always enter a string with a random number attached by `@gmail.com` into input elements? For example, search input?